### PR TITLE
TT-1603: Allow hub-config to return IDPs relevant to a particular RP …

### PR DESCRIPTION
**Pull request by CharlesIC**
…and LoA.

Implemented logic in hub to filter available IDPs on both transaction entity and supported LOA
IdentityProviderConfigEntityData: renamed isOnboarding(loa) to isOnboardingAtLoa(loa); added unit tests for the class

Authors: @CharlesIC, @hugh-emerson

### CODE REVIEW

File: hub/config/src/main/java/uk/gov/ida/hub/config/domain/IdentityProviderConfigEntityData.java [hunk 1]
```diff
@@ -66,6 +65,11 @@ public String getEntityId() {
         return entityId;
     }
 
+    @Override
+    public String toString() {
+        return String.format("%s | %s", this.getEntityId(), this.getSimpleId());
```
Comment (hugh-emerson) [inline hub/config/src/main/java/uk/gov/ida/hub/config/domain/IdentityProviderConfigEntityData.java]: Why did you add this toString method?

File: hub/config/src/test/java/uk/gov/ida/hub/config/domain/filters/IdpPredicateFactoryPredicatesTest.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    public void shouldReturnIdpsForLoaForNonOnboardingTransactionEntity() {
+        final Set<Predicate<IdentityProviderConfigEntityData>> loa1Predicate = idpPredicateFactory
+                .createPredicatesForTransactionEntityAndLoa(transactionEntityNonOnboarding, LevelOfAssurance.LEVEL_1);
+
+        final Set<IdentityProviderConfigEntityData> filteredIdps = getFilteredIdps(allIdps, loa1Predicate);
+
+        final IdentityProviderConfigEntityData[] expectedFilteredIdps = {nonOnboardingLoa1Idp, nonOnboardingAllLevelsIdp,
+                onboardingLoa2Idp, onboardingLoa2IdpOtherOnboardingEntity};
+
+        assertThat(filteredIdps).hasSameSizeAs(expectedFilteredIdps);
```
Comment (hugh-emerson) [inline hub/config/src/test/java/uk/gov/ida/hub/config/domain/filters/IdpPredicateFactoryPredicatesTest.java]: hasSameSizeAs should be unnecessary when you are asserting containsOnly.

From docs for containsOnly:
`Verifies that the actual group contains only the given values and nothing else`

File: hub/config/src/main/java/uk/gov/ida/hub/config/domain/IdentityProviderConfigEntityData.java (hunk 1 above)
Comment (CharlesIC) [reply hub/config/src/main/java/uk/gov/ida/hub/config/domain/IdentityProviderConfigEntityData.java]: Yes, it was for the unit tests as otherwise the entities couldn't be identified in the error messages. Also helpful if we ever get an exception so the useful info would show up in the message instead of the random numeric id generated by JVM

File: hub/config/src/test/java/uk/gov/ida/hub/config/domain/filters/IdpPredicateFactoryPredicatesTest.java (hunk 2 above)
Comment (CharlesIC) [reply hub/config/src/test/java/uk/gov/ida/hub/config/domain/filters/IdpPredicateFactoryPredicatesTest.java]: Just me being extra pedantic.... The extra check can be removed if it's confusing but I don't think it hurts to have it there.
